src/wp-includes/rest-api/class-wp-rest-server.php

```diff
@@ -1371,17 +1371,21 @@ public function remove_header( $key ) {
 	 * @return string Raw request data.
 	 */
 	public static function get_raw_data() {
+		// phpcs:disable PHPCompatibility.Variables.RemovedPredefinedGlobalVariables.http_raw_post_dataDeprecatedRemoved
 		global $HTTP_RAW_POST_DATA;
 
-		/*
-		 * A bug in PHP < 5.2.2 makes $HTTP_RAW_POST_DATA not set by default,
-		 * but we can do it ourself.
-		 */
-		if ( ! isset( $HTTP_RAW_POST_DATA ) ) {
+		if ( isset( $HTTP_RAW_POST_DATA ) ) {
+			_doing_it_wrong( __FUNCTION__, __( '`$HTTP_RAW_POST_DATA` is deprecated in PHP 5.6 and will be removed in 7.0. Use `php://input` instead.' ), '5.5.0' );
```

Comment:
This will also throw a "doing it wrong" when the code is used on PHP 5.6, which I don't think is the intention.